### PR TITLE
added serializer for Stamen layers for MapFish V3

### PR DESCRIPTION
**Pull request by Unraveler**
Adds a serializer for V3 to serialize Stamen layers (toner, terrain, watercolor, labels,...)

Please review @terrestris/devs

### CODE REVIEW

File: src/serializer/MapFishPrintV3StamenSerializer.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+     * @param {Object} opts Additional properties to pass to the serialized
+     *   layer object that can't be obtained by the layer itself. It can also be
+     *   used to override all generated layer values, e.g. the image format.
+     * @return {Object} The serialized/encoded layer.
+     */
+    serialize(layer, opts = {}) {
+        const serialized = super.serialize(layer, opts);
+        const source = layer.getSource();
+
+
```
Comment (dnlkoch) [inline src/serializer/MapFishPrintV3StamenSerializer.js:40]:
```suggestion
```

File: src/serializer/MapFishPrintV3StamenSerializer.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        if (!this.validateSource(source)) {
+            return;
+        }
```
Comment (dnlkoch) [inline src/serializer/MapFishPrintV3StamenSerializer.js:43]: This check can be performed before calling the super method.

File: src/serializer/MapFishPrintV3StamenSerializer.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        serialized.baseURL = source.getUrls()[0];
+        serialized.tileSize = [256, 256]
```
Comment (dnlkoch) [inline src/serializer/MapFishPrintV3StamenSerializer.js:45]: Is this different from the given object? OSM should be in size 256 as well.

**Review by dnlkoch — COMMENTED** [review]
Very nice @Unraveler!

A simple test would be a great addition.

File: src/serializer/MapFishPrintV3StamenSerializer.js (hunk 3 above)
Comment (Unraveler) [reply src/serializer/MapFishPrintV3StamenSerializer.js:44]: Yes, you are correct, i have  seen it wrong. I'll simply remove this

**Comment by Unraveler** [comment]
you can check again  @dnlkoch

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**